docs/work.md

```diff
@@ -0,0 +1,423 @@
+# Working on a project
+
+TODO: Update following tree (and use `tree` fence).
+
+The generated project has this structure:
+
+```
+📁 your_project ------------------- # your freshly created project!
+├── 📄 CHANGELOG.md --------------- # 
+├── 📄 CODE_OF_CONDUCT.md --------- # 
+├── 📁 config --------------------- # tools configuration files
+│   ├── 📄 coverage.ini ----------- # 
+│   ├── 📄 mypy.ini --------------- # 
+│   ├── 📄 pytest.ini ------------- # 
+│   └── 📄 ruff.toml -------------- # 
+├── 📄 CONTRIBUTING.md ------------ # 
+├── 📁 docs ----------------------- # documentation pages
+│   ├── 📄 changelog.md ----------- # 
+│   ├── 📄 code_of_conduct.md ----- # 
+│   ├── 📄 contributing.md -------- # 
+│   ├── 📄 credits.md ------------- # 
+│   ├── 📁 css -------------------- # extra CSS files
+│   │   ├── 📄 material.css ------- # 
+│   │   └── 📄 mkdocstrings.css --- # 
+│   ├── 📄 index.md --------------- # 
+│   └── 📄 license.md ------------- # 
+├── 📄 duties.py ------------------ # the project's tasks
+├── 📄 LICENSE -------------------- # 
+├── 📄 Makefile ------------------- # for auto-completion (it calls scripts/make)
+├── 📄 mkdocs.yml ----------------- # docs configuration
+├── 📄 pyproject.toml ------------- # project metadata and dependencies
+├── 📄 README.md ------------------ # 
+├── 📁 scripts -------------------- # helper scripts
+│   ├── 📄 gen_credits.py --------- # script to generate credits
+│   ├── 📄 gen_ref_nav.py --------- # script to generate code reference nav
+│   └── 📄 make ------------------- # a convenience script to run tasks
+├── 📁 src ------------------------ # the source code directory
+│   └── 📁 your_package ----------- # your package
+│       ├── 📄 cli.py ------------- # the command line entry point
+│       ├── 📄 __init__.py -------- # 
+│       ├── 📄 __main__.py -------- # 
+│       └── 📄 py.typed ----------- # 
+└── 📁 tests ---------------------- # the tests directory
+    ├── 📄 conftest.py ------------ # pytest fixtures, etc.
+    ├── 📄 __init__.py ------------ # 
+    └── 📄 test_cli.py ------------ # 
+```
+
+## Environment
+
+The project is configured to use [direnv](https://direnv.net/).
+If direnv is loaded in your shell, allow it in the project with
+`direnv allow`. It will add the `scripts` folder to your PATH
+when you enter the repository (and remove it when you exit it).
+The `scripts` folder has a `make` Bash script in it:
+it will shadow any `make` command you have in your PATH
+(this is indented!).
+
+If you don't have or don't use direnv, you can still use
+the official `make` command, though you won't be able
+to pass arguments to some of the actions.
+
+In the rest of the documentation, we will use `make` commands,
+but you can also directly call `scripts/make`.
+
+See [Tasks](#tasks) to learn more.
+
+## Dependencies
+
+TODO: Document dependency management.
+
+## Tasks
+
+TODO: Design task system and document it.
```

Comment:
What does `#1` refer to? PR number? I generally don't use a scope for todos because I never found great interest for those, but please enlighten me :slightly_smiling_face:

Reply:
Any benefit, apart from historical pointers? I prefer using references to commit themselves rather than vendor-locked-in resources such as GitHub PRs. And anyway, these TODOs will be removed once we process them, and the commits will probably be squashed.

Reply:
> And anyway, these TODOs will be removed once we process them, and the commits will probably be squashed.
Ahh! Nevermind then!